ocs_ci/ocs/platform_nodes.py

```diff
@@ -556,6 +538,142 @@ def restart_nodes_teardown(self):
         if stopped_instances:
             self.aws.start_ec2_instances(instances=stopped_instances, wait=True)
 
+    def create_nodes(self, node_conf, node_type, num_nodes):
+        """
+        create aws instances of nodes
+
+        Args:
+            node_conf (dict): of node configuration
+            node_type (str): type of node to be created RHCOS/RHEL
+            num_nodes (int): Number of node instances to be created
+
+        Returns:
+           list: of AWSUPINode objects
+
+        """
+        node_list = []
+        node_cls = self.nodes_map[
+            f'{self.platform.upper()}{self.deployment_type.upper()}Node'
+        ]
+
+        if node_type.upper() == 'RHCOS':
+            workers_stacks = self.aws.get_worker_stacks()
+            logger.info(f"Existing worker stacks: {workers_stacks}")
+            existing_indexes = self.get_existing_indexes(workers_stacks)
+            logger.info(f"Existing indexes: {existing_indexes}")
+            slots_available = self.get_available_slots(existing_indexes, num_nodes)
+            logger.info(f"Available indexes: {slots_available}")
+            for slot in slots_available:
+                node_id = slot
+                node_list.append(node_cls(node_conf, node_type))
+                node_list[-1]._prepare_node(node_id)
+        elif node_type.upper() == 'RHEL':
+            rhel_workers = len(get_typed_worker_nodes('rhel'))
+            for i in range(num_nodes):
+                node_id = i + rhel_workers
+                node_list.append(node_cls(node_conf, node_type))
+                node_list[-1]._prepare_node(node_id)
+
+        # Make sure that csr is approved for all the nodes
+        # not making use of csr.py functions as aws rhcos has long
+        # delays for csr to appear
+        if node_type.upper() == 'RHCOS':
+            self.approve_all_nodes_csr(node_list)
+
+        return node_list
+
+    @retry(
+        (exceptions.PendingCSRException, exceptions.TimeoutExpiredError),
+        tries=4,
+        delay=10,
+        backoff=1
+    )
+    def approve_all_nodes_csr(self, node_list):
```

Comment:
Is this related only to AWS nodes or do you thing the same will be used for vmware?  I was thinking to move it to csr module but then I see how you are using those different function across this class so it will be harder to make some nice ordering.  You can keep it here for now but most likely we will need to move this to base class and reuse for vmware? Or found better way.

Reply:
@petr-balogh , Based on https://github.com/red-hat-storage/ocs-ci/issues/1941 discussion, for vmware different condition is working which wasn't for AWSUPI platform. Hence I just wrote a simple wrapper function around existing csr library function where I am relying on node to be listed in ```get nodes``` output.  I will talk to Vijay if we can reconcile on this.